src/Microsoft.CodeAnalysis.Analyzers/UnitTests/FixAnalyzers/FixerWithFixAllAnalyzerTests.cs

```diff
@@ -90,11 +90,13 @@ public class MyDerivedCodeActionWithEquivalenceKey : MyAbstractCodeActionWithEqu
             }
 
             // Verify expected diagnostics for fixer that supports FixAllProvider.
-            await VerifyCS.VerifyAnalyzerAsync(source + fixAllProviderString + sourceSuffix, compilerDiagnostics, expected);
+            await VerifyCS.VerifyAnalyzerAsync(source + fixAllProviderString + sourceSuffix, expected);
 
             // Verify RS1016 (OverrideGetFixAllProviderRule) diagnostic for fixer that does not support FixAllProvider.
-            expected = new DiagnosticResult[] { missingGetFixAllProviderOverrideDiagnostic };
-            await VerifyCS.VerifyAnalyzerAsync(source + sourceSuffix, compilerDiagnostics, expected);
+            var allExpected = new[] { missingGetFixAllProviderOverrideDiagnostic }
+                .Concat(expected.Where(x => x.Severity == DiagnosticSeverity.Error))
```

Comment:
Not sure whether that's best to check the severity or something else. We could check ID starts with `CS` (or `BC` for vbnet).

Reply:
@sharwell Is it ok for you?